http/dotnet/httpclient/src/HttpClientBuilder.cs

```diff
@@ -59,7 +60,27 @@ public static DelegatingHandler ChainHandlersCollectionAndGetFirstLink(params De
                     previousHandler.InnerHandler = handler;
                 }
             }
+            if(finalHandler != null)
+                handlers[^1].InnerHandler = finalHandler;
             return handlers.First();
         }
+        /// <summary>
+        /// Creates a <see cref="DelegatingHandler"/> to use for the <see cref="HttpClient" /> from the provided <see cref="DelegatingHandler"/> instances. Order matters.
+        /// </summary>
+        /// <param name="handlers">The <see cref="DelegatingHandler"/> instances to create the <see cref="DelegatingHandler"/> from.</param>
+        /// <returns>The created <see cref="DelegatingHandler"/>.</returns>
+        public static DelegatingHandler ChainHandlersCollectionAndGetFirstLink(params DelegatingHandler[] handlers)
```

Comment:
Note: This is added as using the `params` keyword together with optional arguments sort of negates the concept of optional parameters. 
The other parameters not covered by the params keyword sort of become mandatory from the point of view of the caller of the function as they still have to specify it due to the "params" after it).

Reply:
do you mean you could simply have a method `public static DelegatingHandler ChainHandlersCollectionAndGetFirstLink(HttpMessageHandler finalHandler = default, params DelegatingHandler[] handlers)` ?

Reply:
I mean that even if you do that, the caller will still have to specify the first parameter. That means a caller trying to do this will have their code failing to build.

```cs
ChainHandlersCollectionAndGetFirstLink(defaultHandlers.ToArray()) // does not build.
```
My understanding is that this is caused by the conflict of optional parameters and parameters with `params` keyword both need to be the last parameters in the list of parameters. 
So having them together makes things tricky. 😅 
https://stackoverflow.com/questions/3948971/c-sharp-4-0-optional-parameters-and-params-do-not-work-together

Reply:
yeah ok, we'd have to ask the caller to use named parameters in that case, which makes things clunky when we can simply have an overload. Thasnk!